Tests/Functional/ProfilingTest.php

```diff
@@ -63,14 +64,15 @@ public function testProfilingWithCachePlugin()
         $this->assertEquals('example.com', $stack->getRequestHost());
     }
 
+    /**
+     * @expectedException \Exception
```

Comment:
i discussed once with sebastian bergman and he is saying he agrees that the annotation is less clean than calling setExpectedException. the reason is that if the exception is thrown too early, the test would wrongly be green. this is particularly true when we expect such a generic exception

Reply:
So, it's better to have a if/else at the right position in the test?

Reply:
why if/else? afaik the $this->setExpectedException call is the best practice way. this makes sure the exception is only considered correct when thrown after that line in the code, and you have namespace resolution.

Reply:
`setExpectedException` was removed in 6.0 and replaced by `expectException`. That's why I moved to annotation.

Reply:
Yeah, We need the annotation for BC purpose...

Reply:
We can keep this as-is and switch to `expectException` in next-major.